api/v1beta2/machine_types.go

```diff
@@ -696,7 +696,7 @@ type Bootstrap struct {
 	// dataSecretName is the name of the secret that stores the bootstrap data script.
 	// If nil, the Machine should remain in the Pending state.
 	// +optional
-	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MinLength=0
```

Comment:
consider make this not a pointer in v1beta2 based on the outcome of https://github.com/kubernetes-sigs/cluster-api/pull/12153#discussion_r2077940571

Reply:
I think that should not be part of this PR